library/src/test/java/bio/terra/pfb/utils/CompareOutputUtils.java

```diff
@@ -0,0 +1,142 @@
+package bio.terra.pfb.utils;
+
+import static bio.terra.pfb.utils.CompareOutputUtils.FileExtension.JSON;
+import static bio.terra.pfb.utils.CompareOutputUtils.FileExtension.TXT;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.pfb.PfbReader;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CompareOutputUtils {
+  private static final Logger logger = LoggerFactory.getLogger(CompareOutputUtils.class);
+
+  public static void compareJavaPfbWithPyPfb(
+      String fileName, PfbCommandType commandType, String filePath, FileExtension fileExtension)
+      throws IOException {
+    ObjectMapper mapper = new ObjectMapper();
+    PfbReader reader = new PfbReader();
+    String pythonOutput;
+    try {
+      pythonOutput =
+          Files.readString(
+              Paths.get(getPyPfbOutputFilePath(fileName, commandType, filePath, fileExtension)));
+    } catch (IOException e) {
+      throw new IOException("Error reading file: " + fileName, e);
+    }
+    String avroFilePath = getAvroFilePath(fileName, filePath);
+    String javaPfbOutput = "";
+    switch (commandType) {
+      case SHOW_SCHEMA:
+        javaPfbOutput = reader.showSchema(avroFilePath);
+        break;
+      case SHOW_METADATA:
+        javaPfbOutput = reader.showMetadata(avroFilePath);
+        break;
+      case SHOW_NODES:
+        javaPfbOutput = reader.showNodes(avroFilePath);
+        break;
+      default:
+        throw new InvalidObjectException("Invalid command type");
+    }
+
+    if (fileExtension.equals(JSON)) {
+      JsonNode pyPfbJsonOutput = mapper.readTree(pythonOutput);
+      JsonNode showSchemaOutputJson = mapper.readTree(javaPfbOutput);
+      PfbJsonCompareIgnoringOrder cmp = new PfbJsonCompareIgnoringOrder();
+      assertTrue(pyPfbJsonOutput.equals(cmp, showSchemaOutputJson));
+      //      assertThat(pyPfbJsonOutput, equalTo(showSchemaOutputJson));
+    } else if (fileExtension.equals(TXT)) {
+      assertThat(pythonOutput, equalTo(javaPfbOutput));
+    } else {
+      throw new InvalidObjectException("Invalid file extension");
+    }
+  }
+
+  public static void compareJSONLineByLine(
```

Comment:
There are libraries for writing tests that compare JSON, it might be better to use them here rather than writing your own. The one I've used is [JSONassert](https://github.com/skyscreamer/JSONassert)

Reply:
I'm able to more directly use the Object Mapper to compare JSON for most cases. This is a special case because each line in the file is a json object, rather than an entire json file.

Reply:
Using `JSONAssert` you don't need to use ObjectMapper, and you don't need your `PfbJsonCompareIgnoringOrder` class either. I ran this locally and it seems to work:
```
  public static void compareJavaPfbWithPyPfb(
      String fileName, PfbCommandType commandType, String filePath, FileExtension fileExtension)
      throws IOException {
    PfbReader reader = new PfbReader();
    String pythonOutput =
        Files.readString(
            Paths.get(getPyPfbOutputFilePath(fileName, commandType, filePath, fileExtension)));
    String avroFilePath = getAvroFilePath(fileName, filePath);
    String javaPfbOutput =
        switch (commandType) {
          case SHOW_SCHEMA -> reader.showSchema(avroFilePath);
          case SHOW_METADATA -> reader.showMetadata(avroFilePath);
          case SHOW_NODES -> reader.showNodes(avroFilePath);
          default -> throw new InvalidObjectException("Invalid command type");
        };

    switch (fileExtension) {
      case JSON -> JSONAssert.assertEquals(pythonOutput, javaPfbOutput, true);
      case TXT -> assertThat(pythonOutput, equalTo(javaPfbOutput));
      default -> throw new InvalidObjectException("Invalid file extension");
    }
  }

  public static void compareJSONLineByLine(
      String fileName, PfbCommandType commandType, String filePath) throws IOException {
    PfbReader reader = new PfbReader();

    String avroFilePath = getAvroFilePath(fileName, filePath);
    List<String> javaOutput = reader.show(avroFilePath);
    Path file = Paths.get(getPyPfbOutputFilePath(fileName, commandType, filePath, JSON));
    List<String> lines = Files.readAllLines(file);
    for (int i = 0; i < lines.size(); i++) {
      JSONAssert.assertEquals(lines.get(i), javaOutput.get(i), true);
    }
  }
```

I also made some minor changes here with respect to reading the lines of the file using `Files.readAllLines()` and removed the code that catches `IOException`.

Reply:
Ah, ok! I see that JSON assert provides good error messages when the json doesn't match. 
Thanks!